src/components/app/App.js

```diff
@@ -0,0 +1,206 @@
+import React from 'react';
+import PaymentsDetails from '../payments-details/paymentsDetails';
+import {BrowserRouter, Switch} from 'react-router-dom';
+import MainContent from '../main_content/mainContent'
+
+
+import {Route} from 'react-router-dom';
+
+import './app.css';
+
+
+export default class App extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            users: [],
+            spendings: [],
+
+            sumOfGroupSpent : 0,
+        }
+        ;
+
+        this.onAdd = this.onAdd.bind(this);
+        this.addItem = this.addItem.bind(this);
+        this.getObjOfSpendings = this.getObjOfSpendings.bind(this);
+        this.countTotalSpendings = this.countTotalSpendings.bind(this);
+        this.onRemoveUsers = this.onRemoveUsers.bind(this);
+        this.onRemoveItems = this.onRemoveItems.bind(this);
+        this.onCancel = this.onCancel.bind(this);
+        this.getArrOfUsersSpends = this.getArrOfUsersSpends.bind(this);
+    }
+
+    async onAdd(text) {
+        let id = 1;
+        let newUser;
+        if (this.state.users.length === 0) {
+                newUser = {
+                name: text,
+                id: id,
+                totalSpendings: 0,
+                groupSpengings: 0,
+                finalSpendings: 0
+            };
+        } else {
+                newUser = {
+                name: text,
+                id: this.state.users[this.state.users.length-1].id + 1,
+                totalSpendings: 0,
+                groupSpengings: 0,
+                finalSpendings: 0
+            };
+        }
+        await this.setState({
+            users: [...this.state.users, newUser]
+        });
+    }
+
+    addItem(title) {
+        let id = 1;
+        let newSpendings;
+        if (this.state.spendings.length === 0) {
+            newSpendings = {
+                title: title,
+                amount: 0,
+                id: id
+            }
+        } else {
+                newSpendings = {
+                    title: title,
+                    amount: 0,
+                    id: this.state.spendings[this.state.spendings.length - 1].id + 1
+            }
+        }
+        this.setState(({spendings}) => {
+            let newArr = [...this.state.spendings, newSpendings];
+            return {
+                spendings: newArr
+            }
+        })
+
+    }
+
+    getObjOfSpendings(arr){
```

Comment:
not sure what this function does but it doesn't gets value as one could guess from the name, it sets the value into the local state

lets rename the function so that its easier to follow as per what it does